pkg/source/secret_source.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package source
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var _ source.SyncingSource = &SecretSource{}
+
+// SecretSource is the event source of specified secrets
+type SecretSource struct {
+	secretInformer cache.SharedIndexInformer
+}
+
+// NewImportSecretSource return a SecretSource only for import secrets
+func NewImportSecretSource(secretInformer cache.SharedIndexInformer) *SecretSource {
+	return &SecretSource{secretInformer: secretInformer}
+}
+
+// NewAutoImportSecretSource return a SecretSource only for auto import secrets
+func NewAutoImportSecretSource(secretInformer cache.SharedIndexInformer) *SecretSource {
```

Comment:
It seems duplicate with NewImportSecretSource

Reply:
Just use different name to distinguish different informer